### PR TITLE
Fix setStyle breaking with globe view

### PR DESCRIPTION
**Pull request by avpeery**
When toggling setStyle with globe view, the globe does not get re-draped since `applyProjectionUpdate()` (which calls `this.style.setTerrainForDraping()`) is not called when there isn't a change in the projection name. This change add a call to `this.style.setTerrainForDraping()` in the case that a new style is reset with the same globe view projection as the previous style.

Before: 

https://user-images.githubusercontent.com/42715836/170098426-cde7d78e-861e-48ec-9d3e-d6a3a4e50134.mov

After:

https://user-images.githubusercontent.com/42715836/170098454-19b5261a-1c1b-4719-92e5-73f1fc136bd2.mov

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] write tests for all new functionality
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

**Comment by avpeery** [comment]
This PR is a "quick fix" solution; however, the origin of the problem also stems from #11916. Removing rewriting the stylesheet introduced [here](https://github.com/mapbox/mapbox-gl-js/pull/11204/files#diff-c19065149b8e6b02ea30bd2576429dfc2f917695da9382c77a6957401f6fef4eR347) resolves the draping issue

File: src/ui/map.js [hunk 1]
```diff
@@ -1200,6 +1200,8 @@ class Map extends Camera {
             }
             this.style.applyProjectionUpdate();
             this._update(true);
+        } else if (this.transform.projection.requiresDraping && (!this.getTerrain() || !this.style.stylesheet.terrain)) {
```
[the remark this answers is not in the document]
Comment (avpeery) [reply src/ui/map.js:1203]: Yes, it avoids calling terrain again - I used the same logic from line https://github.com/mapbox/mapbox-gl-js/blob/main/src/style/style.js#L385 to determine if draping should be set. Noted on the cosmetic improvement!

[the remark this answers is not in the document]
Comment (SnailBones) [reply src/ui/map.js]: The line you mention can be rewritten as follows:
```js
!(this.getTerrain() || this.stylesheet.terrain);
```
which is different from
```js
!this.getTerrain() || !this.stylesheet.terrain
```

For instance ,when `this.getTerrain()` returns `true` and `this.stylesheet.terrain` is `false`, the former becomes: 
```js
!(true || false)
```
```js
!(true)
```
```js
false
```
In the same case, the latter becomes:
```js
!true || !false
```
```js
false || true
```
```js
true
```

So I think that this code is applying the update in some cases where it is unnecessary.

[the remark this answers is not in the document]
Comment (avpeery) [reply src/ui/map.js]: Thanks for bringing my attention to this @SnailBones! fixed

**Review by SnailBones — APPROVED** [review]
Nice work 👍

**Review by karimnaaji — APPROVED** [review]
LGTM, nice find!

**Pull request merged**